File: WeActive-SPA/src/app/_models/participant.ts
import { Activity } from './activity';
import { User } from './user';

export interface Participant {
    activityId: number;
    activity: Activity;
    userId: number;
    user: User;
    participantStatus: ParticipantStatus;
}

export enum ParticipantStatus {
    Joined,
    Interested,
    Unconfirmed,
    Resigned
}

File: WeActive.API/Models/Participant.cs
namespace WeActive.API.Models
{
    public class Participant
    {
        public int ActivityId { get; set; }
        public Activity Activity { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public ParticipantStatus ParticipantStatus { get; set; }
    }

    public enum ParticipantStatus { Joined, Interested, Unconfirmed, Resigned}
}